index.d.ts

```diff
@@ -0,0 +1,20 @@
+export class ManyKeysWeakMap<K extends object, V> {
```

Comment:
Thank you for the PR!

ManyKeysWeakMaps should extend WeakMap:

https://github.com/fregante/many-keys-weakmap/blob/a887ee360e26d50cf82f6f9a9a5c7c952abc4e48/index.js#L31

Reply:
Also I pushed some meta updates to `master` to ensure this file is linted, I think it doesn't pass eslint at the moment

Reply:
Also related module:

- https://github.com/fregante/many-keys-map/issues/2

Reply:
I'm not sure what extending weakmap would actually do here, are there other methods/properties that this class doesn't override?

Reply:
The current type doesn't work with `instanceof WeakMap`, but it should

Reply:
It looks like the types should just be something like:

```ts
export class X<K extends unknown[], V = unknown> extends WeakMap<K, V> {}
```

See: https://www.typescriptlang.org/play?target=2&ssl=1&ssc=74&pln=1&pc=1#code/KYDwDg9gTgLgBAYwDYEMDOa4A0A8BpOUGYAOwBNMBXEgaxIgHcSBtAXQBo4A1OAXjmp1GJAHyEQxcpgDqwFDQCyKMPk5cxAbwC+AKB0IIJNPBB84JYA2w4SlALYAjYFDYiAFAEodIAHRpgMG7MAIycAEwccADMHgDc3n4BbsFwAPQAVIRQUNBw6amcMbFAA

Reply:
Ok I see, good idea.

I've changed it to that approach. I used `K extends object` and simply `V` (no default) to match the WeakMap generic API.

Reply:
Using `extends object` makes the linter shout at me, but I think it's the right choice to match the native WeakMap types.